opm/simulators/timestepping/TimeStepControl.cpp

```diff
@@ -141,17 +142,23 @@ namespace Opm
         errors_[ 2 ] = error;
         for( int i=0; i<2; ++i ) {
             assert(std::isfinite(errors_[i]));
-            assert(errors_[i]>0);
+            assert(errors_[i]>=0);
         }
 
-        if( error > tol_ )
+        if( errors_[2] > tol_ )
         {
             // adjust dt by given tolerance
             const double newDt = dt * tol_ / error;
             if( verbose_ )
                 std::cout << "Computed step size (tol): " << unit::convert::to( newDt, unit::day ) << " (days)" << std::endl;
             return newDt;
         }
+        else if (errors_[1] * errors_[2] == 0.)
+        {
+            if ( verbose_ )
+                std::cout << "The solution between time step does not change, there is no time step constraint from the PID time step control " << std::endl;
```

Comment:
It's tangential to the theme of the PR, but do you think we can switch these to using the logging system instead?  Messages output to `cout` will just vanish unless the user takes extra care to capture a log file.

Reply:
Yes, we can. I will do that.  I did not do it because it was not the initial attention for this PR. It will be addressed.